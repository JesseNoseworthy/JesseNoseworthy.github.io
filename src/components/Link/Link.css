@import '../../styles/variables.css';
@import '../../styles/mixins.css';

:local(.Link) {

}