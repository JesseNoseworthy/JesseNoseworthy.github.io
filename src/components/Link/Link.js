import styled from 'styled-components';

const Link = styled.a`
  text-decoration: none;
`;

export default Link;