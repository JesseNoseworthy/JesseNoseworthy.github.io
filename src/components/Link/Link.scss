// All Link styles go here

a {

}