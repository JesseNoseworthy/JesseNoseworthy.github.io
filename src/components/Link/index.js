import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import styles from './Link.css';
import { isExternalLink } from '../../utils';

const Link = ({
  className,
  variant,
  href,
  children,
  isNav,
  isWhite,
  currentSegment,
  ...others
}) => (
    <a  href={href}
      className={classNames(styles.Link, variant, className)}
      target={isExternalLink(href) ? '_blank' : '_self'}
      {...others}>
    {children}
  </a>
);

Link.propTypes = {
  variant: PropTypes.string,
  className: PropTypes.string,
  href: PropTypes.string,
  children: PropTypes.any,
  isNav: PropTypes.bool,
};

Link.defaultProps = {
  isNav: false,
}

export default Link;