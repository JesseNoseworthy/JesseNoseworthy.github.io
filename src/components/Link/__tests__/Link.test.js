import React from 'react';
import Link from '../index';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';

describe('Link', () => {
  describe('Shallow rendering', () => {
    it('renders as a button element', function() {
      expect(shallow(<Link href='google.com' />).type()).toEqual('a');
    });
  });

  describe('Mounting', () => {
    it('allows us to set props', () => {
      const wrapper = mount(<Link href='' />);

      wrapper.setProps({ href: 'google.com' });
      expect(wrapper.props().href).toEqual('google.com');
    });

    it('simulates click events', () => {
      const handleClick = jasmine.createSpy();
      const wrapper = mount(<Link href='google.com' onClick={handleClick} />);
      wrapper.find('a').simulate('click');
      expect(handleClick).toHaveBeenCalled();
    });
  });

  describe('Snapshot tests', () => {
    it('default state (only required props)', () => {
      const component = renderer.create(
        <Link href='google.com' />
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('sets children as text', function() {
      const component = renderer.create(
        <Link href='google.com'>Test</Link>
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('parses and sets innerHtml', function() {
      const component = renderer.create(
        <Link href='google.com' innerHtml={'Test<sup>TM</sup>'}/>
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
