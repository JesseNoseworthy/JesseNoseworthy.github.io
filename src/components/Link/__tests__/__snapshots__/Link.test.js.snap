exports[`Link Snapshot tests default state (only required props) 1`] = `
<a
  className=""
  href="google.com" />
`;

exports[`Link Snapshot tests parses and sets innerHtml 1`] = `
<a
  className=""
  href="google.com"
  innerHtml="Test<sup>TM</sup>" />
`;

exports[`Link Snapshot tests sets children as text 1`] = `
<a
  className=""
  href="google.com">
  Test
</a>
`;
